modules/RTC/TPCUtils.js

```diff
@@ -88,40 +87,51 @@ export class TPCUtils {
      */
     _calculateActiveEncodingParams(localVideoTrack, codec, newHeight) {
         const codecBitrates = this.codecSettings[codec].maxBitratesVideo;
-        const height = localVideoTrack.getHeight();
-        const desktopShareBitrate = this.pc.options?.videoQuality?.desktopbitrate || DESKTOP_SHARE_RATE;
+        const trackHeight = localVideoTrack.getHeight();
+        const effectiveNewHeight = newHeight > trackHeight ? trackHeight : newHeight;
```

Comment:
Not sure I'm following this right. Here we are picking `newHeight` only if it's _smaller_.

So what happens if we are currently only sending the middle layer and someone starts requesting the high layer? Won't the effective new height be the previous value here?

Since we use the new height to compute the active value, it feels weird we might use the current value for computing the bitrate.

Reply:
🙏  I think it will help those not working with the code as closely as you do.